bundles/framework/publisher2/tools/PanButtonsTool.js

```diff
@@ -43,6 +53,63 @@ Oskari.clazz.define('Oskari.mapframework.publisher.tool.PanButtonsTool',
             } else {
                 return null;
             }
+        },
+        /**
+         * Get extra options.
+         * @method getExtraOptions
+         * @public
+         *
+         * @returns {Object} jQuery element
+         */
+        getExtraOptions: function () {
+            var me = this;
+            if (!me._extraOptions) {
+                const initialConf = me._getToolPluginMapfullConf();
+                const showArrows = initialConf && initialConf.config && initialConf.config.showArrows;
+                const extraOptions = me._templates.extraOptions.clone();
+                extraOptions.find('.arrows-selection label span').append(me.__loc.panButtonsOptions.showArrows);
+                const arrowsCheckbox = extraOptions.find('.arrows-selection label input')
+                    .on('change', function () {
+                        const isChecked = jQuery(this).is(':checked');
+                        me.__plugin.setShowArrows(isChecked);
+                    });
+                if (showArrows) {
+                    arrowsCheckbox.prop('checked', true).change();
+                    me.__plugin.setShowArrows(true);
+                }
+                me._extraOptions = extraOptions;
+            }
+            return me._extraOptions;
+        },
+        /**
+         * @private @method _getToolPluginMapfullConf
+         * Get map view cofiguration (from mapfull) for this tool
+         * @return {Object / null} config or null if not found
+         */
+        _getToolPluginMapfullConf: function () {
+            const { configuration } = this.data || {};
+            if (!configuration) {
+                return null;
+            }
+            const { mapfull = {} } = configuration;
+            const { conf = {} } = mapfull;
+            const { plugins = [] } = conf;
+            // data.configuration.mapfull.conf.plugins
+            return plugins.find(plug => plug.id === 'Oskari.mapframework.bundle.mapmodule.plugin.PanButtons');
+        },
+        init: function (data) {
+            var me = this;
+            me.data = data;
+
+            if (data.configuration && data.configuration.mapfull && data.configuration.mapfull.conf && data.configuration.mapfull.conf.plugins) {
+                const plugins = data.configuration.mapfull.conf.plugins;
+                plugins.forEach(function (plugin) {
+                    if (me.getTool().id !== plugin.id) {
+                        return;
+                    }
+                    me.setEnabled(true);
```

Comment:
Does this work when editing a previously published map so that it restores the arrow options from previous state or do you need to do it here? It might work since you are using the plugin config, but just want to know if it's been tested as it's one of the most common problems with publisher tools :)

Reply:
Should work correctly.